lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -155,6 +155,22 @@ EmitOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
   return success();
 }
 
+LogicalResult
+FetchOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
+                          ValueRange operands, DictionaryAttr attributes,
+                          OpaqueProperties properties, RegionRange regions,
+                          llvm::SmallVectorImpl<Type> &inferredReturnTypes) {
+  Value input = operands[0];
+
+  TypeRange leftFieldTypes = cast<TupleType>(input.getType()).getTypes();
+
+  auto resultType = TupleType::get(context, leftFieldTypes);
+
+  inferredReturnTypes = SmallVector<Type>{resultType};
```

Comment:
The way how this interface is intended is to fill the `SmallVector` that is passed in, so you'd `inferredReturnTypes.push_back(resultType)`. In this case, the function is probably always being called with an empty vector but I think it's still better to stick to the convention.